autoload/utils/common.vim

```diff
@@ -18,39 +18,75 @@ endfunction
 " }}} Private functions "
 
 " Executes commands
-function! utils#common#executeCommands(cmds, open_result, ...) abort
-    let l:cwd = get(a:, 1, getcwd())
-    let l:errFormat = get(a:, 2, '')
-
+function! utils#common#executeCommands(cmds, open_result) abort
+    " add all fields
+    let l:commands = []
+    for l:cmd in a:cmds
+        let l:dict_cmd = {'cmd': l:cmd['cmd']}
+        if !has_key(l:cmd, 'cwd')
+            let l:dict_cmd['cwd'] = getcwd()
+        else
+            let l:dict_cmd['cwd'] = l:cmd['cwd']
+        endif
+        if !has_key(l:cmd, 'errFormat')
+            let l:dict_cmd['errFormat'] = ''
+        else
+            let l:dict_cmd['errFormat'] = l:cmd['errFormat']
+        endif
+        let l:commands += [l:dict_cmd]
+    endfor
     if (g:cmake_build_executor ==# 'dispatch') || (g:cmake_build_executor ==# '' && exists(':Dispatch'))
         " Close quickfix list to discard custom error format
         silent! cclose
+        let l:cmd_line = ''
+        let l:pcwd = getcwd()
+        let l:errFormat = ''
         " Dispatch doesn't support pool of tasks
-        let l:cmd = join(a:cmds, ' && ')
-        if l:cwd != getcwd()
-            let l:cmd = 'cd ' . utils#fs#fnameescape(l:cwd) . ' && ' . l:cmd . ' && cd ' . utils#fs#fnameescape(getcwd())
+        for l:cmd in l:commands
+            " generate common command
+            if l:cmd_line !=# ''
+                let l:cmd_line .= ' && '
+            endif
+            let l:cwd = l:cmd['cwd']
+            if l:cwd['errFormat'] !=# ''
+                let l:errFormat = l:cwd['errFormat']
+            endif
```

Comment:
This `if` isn't necessary, you can always assign since `''` is the default value.
```
let l:errFormat = l:cwd['errFormat']
```

If it is equal to `''`, nothing changes since `l:errFormat` is already `''`.
If it is equal to `'something'`, it will get assigned.

Reply:
Yes, but here I have a cycle to generate one common command. 
Dispatch doesn't support multiple error formats and in this case I would like to save the last errorFormat.

Reply:
Ah, I see. That clears it up, sorry for the noise!